styx-common/src/main/java/com/spotify/styx/model/EventVisitor.java

```diff
@@ -37,8 +38,9 @@
   R triggerExecution(@Getter WorkflowInstance workflowInstance, Trigger trigger);
   R info(@Getter WorkflowInstance workflowInstance, Message message);
   R dequeue(@Getter WorkflowInstance workflowInstance);
-  R submit(@Getter WorkflowInstance workflowInstance, ExecutionDescription executionDescription);
-  R submitted(@Getter WorkflowInstance workflowInstance, String executionId);
+  R submit(@Getter WorkflowInstance workflowInstance, ExecutionDescription executionDescription,
+      @Nullable String executionId);
```

Comment:
Both `executionId` and `executionDescription` are (already) fields in `RunState.data()`. Adding `executionId` to `ExecutionDescription` would be redundant. Moving `executionId` from `RunState.data()` to `ExecutionDescription` could be an option, but not sure we would want that. Either way it would be a bigger and backwards incompatible change.

`RunSpec` is not part of `RunState`. It is only a part of the protocol between `DockerRunner` and `DockerRunnerHandler`.

Although `ExecutionDescription` and `RunSpec` are almost identical, I think they might live in separate bounded contexts and that DRY might not apply here. E.g. we would not want to add `trigger` to `ExecutionDescription` just because `DockerRunner` needs it and it is currently in `RunSpec`. Anyway, this might be a discussion worth having outside this PR.